### PR TITLE
Fixes #408 - implemented clone flag on ovirt_vm resource

### LINKED ISSUE #408: Implement `clone` on `ovirt_vm`

### PR DESCRIPTION
**Pull request by engelmi**
## Please describe the change you are making

This PR fixes #408 and implements the clone flag on the `ovirt_vm` resource. 

## Are you the owner of the code you are sending in, or do you have permission of the owner?

yes

## The code will be published under the BSD 3 clause license. Have you read and understood this license?

yes

### CODE REVIEW

File: internal/ovirt/resource_ovirt_vm_test.go [hunk 1]
```diff
@@ -816,3 +816,116 @@ resource "ovirt_vm" "test" {
 		)
 	}
 }
+
+func TestClone(t *testing.T) {
```
[the remark this answers is not in the document]
Comment (janosdebugs) [reply internal/ovirt/resource_ovirt_vm_test.go:820]: This behavior is correct, the template ID is set to the blank template ID when you clone.

File: internal/ovirt/resource_ovirt_vm.go [hunk 2]
```diff
@@ -164,6 +164,20 @@ var vmSchema = map[string]*schema.Schema{
...
 		Description: "Enable or disable the serial console.",
 	},
+	"clone": {
+		Type:        schema.TypeBool,
+		Optional:    true,
+		Description: "If true, the VM is cloned from the template instead of linked. As a result, the template can be removed and the VM still exists.",
+	},
+}
+
+func cpuModeValues() []string {
```
Comment (janosdebugs) [inline internal/ovirt/resource_ovirt_vm.go:174]: Why is this code in this PR?

**Review by janosdebugs — CHANGES_REQUESTED** [review]
I would suggest adding an extra computed field called `effective_template_id` and only update that.

File: internal/ovirt/resource_ovirt_vm.go (hunk 2 above)
Comment (engelmi) [reply internal/ovirt/resource_ovirt_vm.go:174]: Just moved this function up so all `xyzValues()` are close together.

**Review by janosdebugs — APPROVED** [review]

**Pull request merged**